jax/_src/numpy/fft.py

```diff
@@ -78,7 +78,18 @@ def _fft_core(func_name, fft_type, a, s, axes, norm):
     else:
       s = [a.shape[axis] for axis in axes]
 
-  transformed = lax.fft(a, fft_type, s)
+  # It's faster to compute a real->complex FFT and form the second half by
+  # complex conjugation than to promote to complex and perform a
+  # complex->complex FFT.
+  if (fft_type == xla_client.FftType.FFT
+      and not jnp.issubdtype(a.dtype, np.complexfloating)
+      and len(axes) == 1):
+    # TODO(phawkins): extend this logic to n > 1 dimensions.
+    t = lax.fft(a, xla_client.FftType.RFFT, s)
+    start = -2 + (a.shape[-1] % 2)
+    transformed = jnp.concatenate([t, jnp.conj(t[..., start:0:-1])], axis=-1)
```

Comment:
are you sure this logic is right for both even and odd sizes? I think there might be an off-by-one error, e.g., when the last axis has length 8